js/date-filter.js

```diff
@@ -125,6 +146,127 @@ DateFilter.prototype.handleModifyEvent = function(e, opts) {
+  var dateBegin = this.$minDate.val().split('/');
+  var dateEnd = this.$maxDate.val().split('/');
+  var dateRangeFirst = this.$grid.find('.date-range-first');
+  var dateRangeSecond = this.$grid.find('.date-range-second');
+  var minDateMonth = dateBegin[0];
+  var minDateYear = dateBegin[2];
+  var maxDateMonth = dateEnd[0];
+  var maxDateYear = dateEnd[2];
+  var transactionYear = $('#two-year-transaction-period').val();
+  var $dateBegin;
+  var $dateEnd;
+
+  // the transaction year starts with the later, so begin before
+  dateRangeFirst.find('.date-range-year').html(parseInt(transactionYear - 1));
+  dateRangeFirst.find('ul').attr('data-year', parseInt(transactionYear - 1));
+
+  dateRangeSecond.find('.date-range-year').html(transactionYear);
+  dateRangeSecond.find('ul').attr('data-year', transactionYear);
+
+  // get the elements of the beginning and ending dates
+  $dateBegin = this.$grid.find('ul[data-year="' + minDateYear + '"] ' +
+    'li[data-month="' + minDateMonth + '"]').addClass('begin');
+  $dateEnd = this.$grid.find('ul[data-year="' + maxDateYear + '"] ' +
+    'li[data-month="' + maxDateMonth + '"]').addClass('end');
+
+  // set the selected date range in the grid
+  this.handleDateGridRange($dateBegin, $dateEnd);
+};
+
+DateFilter.prototype.handleDateGridRange = function($dateHoverBegin, $dateHoverEnd) {
+  this.$grid.find('li').removeClass();
+
+  $dateHoverBegin.addClass('selected begin');
+  $dateHoverEnd.addClass('selected end');
+
+  if (!$dateHoverBegin.is($dateHoverEnd)) {
+    $dateHoverBegin.nextUntil('.end').addClass('selected');
+    $dateHoverEnd.prevUntil('.begin').addClass('selected');
+  }
+};
+
+DateFilter.prototype.handleMinDateSelect = function() {
+  var self = this;
+  var $dateBegin = this.$grid.find('.begin');
+  var $dateEnd = this.$grid.find('.end');
+
+  this.$grid.show().removeClass('pick-max').addClass('pick-min');
+  this.$grid.find('.active').removeClass('active');
+  $dateBegin.addClass('active');
+
+  this.$grid.find('li').hover(
+    function() {
+      var dateBeginNum = parseInt($(this).parent().attr('data-year') + $(this).attr('data-month'));
+      var dateEndNum = parseInt($dateEnd.parent().attr('data-year') + $dateEnd.attr('data-month'));
+
+      if (dateBeginNum <= dateEndNum) {
+        self.$grid.removeClass('invalid');
+        self.handleDateGridRange($(this), $dateEnd, $dateBegin, $dateEnd);
+      }
+      else {
+        self.$grid.addClass('invalid');
+      }
+    },
+    function() {
+      self.handleDateGridRange($dateBegin, $dateEnd);
+      $dateBegin.addClass('active');
+    }
+  );
+};
+
+DateFilter.prototype.handleMaxDateSelect = function() {
+  var self = this;
+  var $dateBegin = this.$grid.find('.begin');
+  var $dateEnd = this.$grid.find('.end');
+
+  this.$grid.show().removeClass('pick-min').addClass('pick-max');
+  this.$grid.find('.active').removeClass('active');
+  $dateEnd.addClass('active');
+
+  this.$grid.find('li').hover(
```

Comment:
Do you think this will be an issue adding the event handlers here inside an event handler? I don't know what's best practice in this case.